src/Helpers/CacheMap.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Backpack\Basset\Helpers;
+
+use Illuminate\Filesystem\FilesystemAdapter;
+use Illuminate\Support\Facades\File;
+use Illuminate\Support\Facades\Storage;
+
+class CacheMap
+{
+    private $map = [];
+    private $path;
+    private $isActive = false;
+    private $isDirty = false;
+
+    public function __construct()
+    {
+        $this->isActive = config('backpack.basset.cache_map', false);
+        if (! $this->isActive) {
+            return;
+        }
+
+        /** @var FilesystemAdapter */
+        $disk = Storage::disk(config('backpack.basset.disk'));
+        $this->path = $disk->path('.basset');
+
+        // Load map
+        $this->map = File::exists($this->path) ? json_decode(File::get($this->path), true) : [];
+    }
+
+    /**
+     * Saves the cache map to the .basset file.
+     *
+     * @return void
+     */
+    public function save(): void
+    {
+        if (! $this->isDirty || ! $this->isActive) {
+            return;
+        }
+
+        // save file
+        File::put($this->path, json_encode($this->map, JSON_PRETTY_PRINT));
+    }
+
+    /**
+     * Adds an asset to the cache map.
+     *
+     * @param  string  $asset
+     * @param  string  $path
+     * @return void
+     */
+    public function addAsset(string $asset, string|bool $path = true): void
```

Comment:
I notice the assets get their ABSOLUTE paths added:

<img width="1512" alt="CleanShot 2023-04-21 at 11 45 54@2x" src="https://user-images.githubusercontent.com/1032474/233590288-d07e40ad-0b54-4d21-88a5-f0479c2f6a1b.png">

Doesn't this mean that... if something changes, like the protocol (HTTP/HTTPS) or app name (unlikely)... the whole bassets cache file is useless? Is that a real problem... or not? 🤷‍♂️

Reply:
Good question!
I noticed that, thought about and decided to keep it that way because the .basset file is local, so it's unlikely anything on the machine changes, and that is the default way, to remove the url would "cost" cpu time on each request to add the url again BUT it's irrelevant anyway.

I'll do it because it's safer that way 👌

Reply:
Done!